thirdparty/pcre2/src/sljit/allocator_src/sljitExecAllocatorApple.c

```diff
@@ -104,6 +104,7 @@ static SLJIT_INLINE void apple_update_wx_flags(sljit_s32 enable_exec)
 #else
 #define SLJIT_MAP_JIT	(0)
 #endif
+#define SLJIT_UPDATE_WX_FLAGS(from, to, enable_exec)
```

Comment:
I don't know if it's a correct fix, or why it doesn't fail on GitHub Actions with Xcode in the first place.

With ioscross, don't fulfill the `TARGET_OS_MAC && !TARGET_OS_IPHONE` condition above (which sounds logical), but then `SLJIT_UPDATE_WX_FLAGS` is never defined.

Reply:
Looks right, as to why it was not detected by GHA - no idea.

We might want to file an issue @ https://github.com/PCRE2Project/pcre2 just in case - if it's a bug it'll be (hopefully) fixed, if not - we might learn what we did wrong (I suspect it's more likely to be option № 1).

Reply:
> this is incorrect; iOS needs an implementation for this function and so the build breaking (or showing a warning that says one is missing) is more accurate.

So maybe not right after all 😩

Source: https://github.com/PCRE2Project/pcre2/pull/251#discussion_r1189657926